### PR TITLE
Vertex input

### PR DESCRIPTION
**Pull request by Qozul**
Teapot time.

### CODE REVIEW

File: Vulkan-Pipeline-Assistant/memoryallocator.cpp [hunk 1]
```diff
@@ -5,9 +5,42 @@
...
 
-MemoryAllocator::~MemoryAllocator() { }
+    VkCommandBufferAllocateInfo allocInfo = {};//m_commandBuffer
+    allocInfo.sType = VK_STRUCTURE_TYPE_COMMAND_BUFFER_ALLOCATE_INFO;
+    allocInfo.commandPool = m_commandPool;
+    allocInfo.level = VK_COMMAND_BUFFER_LEVEL_PRIMARY;
+    allocInfo.commandBufferCount = 1;
+
+    result = m_deviceFuncs->vkAllocateCommandBuffers(window->device(), &allocInfo, &m_commandBuffer);
+    if (result != VK_SUCCESS) qFatal("Failed to allocate command buffer for transfer!");
```
Comment (Kney-Delach) [inline Vulkan-Pipeline-Assistant/memoryallocator.cpp:38]: Checking if vulkan functions returning VK_SUCCESS should be macro'd for good coding practices.

File: Vulkan-Pipeline-Assistant/memoryallocator.cpp [hunk 2]
```diff
@@ -5,9 +5,42 @@
...
+    if (result != VK_SUCCESS) qFatal("Failed to allocate command pool for memory allocator!");
 
-MemoryAllocator::~MemoryAllocator() { }
+    VkCommandBufferAllocateInfo allocInfo = {};//m_commandBuffer
+    allocInfo.sType = VK_STRUCTURE_TYPE_COMMAND_BUFFER_ALLOCATE_INFO;
+    allocInfo.commandPool = m_commandPool;
+    allocInfo.level = VK_COMMAND_BUFFER_LEVEL_PRIMARY;
+    allocInfo.commandBufferCount = 1;
+
+    result = m_deviceFuncs->vkAllocateCommandBuffers(window->device(), &allocInfo, &m_commandBuffer);
```
Comment (Kney-Delach) [inline Vulkan-Pipeline-Assistant/memoryallocator.cpp]: Same as previous comment

**Review by Kney-Delach — APPROVED** [review]
Everything seems great, I left a note about potentially macroing a repeated command.

File: Vulkan-Pipeline-Assistant/memoryallocator.cpp (hunk 1 above)
Comment (Qozul) [reply Vulkan-Pipeline-Assistant/memoryallocator.cpp:38]: Good point, I'll make a note on trello to do this when refactoring to have a proper error handling system.

**Pull request merged**